fast/stages/2-networking-a-peering/main.tf

```diff
@@ -17,13 +17,20 @@
 # tfdoc:file:description Networking folder and hierarchical policy.
 
 locals {
+  _region_tokens = { for v in local._regions : v => split("-", v) }
+  _regions = distinct(concat(
+    [var.regions.primary, var.regions.secondary],
+    values(module.dev-spoke-vpc.subnet_regions),
+    values(module.landing-vpc.subnet_regions),
+    values(module.prod-spoke-vpc.subnet_regions),
+  ))
   custom_roles = coalesce(var.custom_roles, {})
-  l7ilb_subnets = {
-    for env, v in var.l7ilb_subnets : env => [
-      for s in v : merge(s, {
-        active = true
-        name   = "${env}-l7ilb-${s.region}"
-    })]
+  region_shortnames = {
+    for k, v in local._region_tokens : k => join("", [
+      substr(v.0, 0, 1),
+      substr(v.1, 0, 1),
+      substr(strrev(v.1), 0, 1),
+    ])
```

Comment:
I really hate this. Lots of magic numbers and hard to follow. See proposal above

Reply:
I really hate regexps lol